ulib/LowStar.Vector.fst

```diff
@@ -497,6 +498,26 @@ let flush #a vec ia i =
   B.free vs;
   Vec fsz asz fvs
 
+val shrink:
+  #a:Type -> vec:vector a ->
+  new_size:uint32_t{new_size <= size_of vec} ->
+  HST.ST (vector a)
+    (requires (fun h0 ->
+      live h0 vec /\ freeable vec /\
+      HST.is_eternal_region (frameOf vec)))
+    (ensures (fun h0 fvec h1 ->
+      frameOf vec = frameOf fvec /\
+      hmap_dom_eq h0 h1 /\
+      live h1 fvec /\ freeable fvec /\
+      modifies (loc_union (loc_addr_of_vector vec)
+                          (loc_vector fvec)) h0 h1 /\
```

Comment:
Why the modifies clause? This code is actually pure, right?

Reply:
Good point! It's just there because I copied it from `flush`; removed now.